docs/en/build.md

```diff
@@ -11,4 +11,36 @@ NodeMCU "application developers" just need a ready-made firmware. There's a [clo
 Occasional NodeMCU firmware hackers don't need full control over the complete tool chain. They might not want to setup a Linux VM with the build environment. Docker to the rescue. Give [Docker NodeMCU build](https://hub.docker.com/r/marcelstoer/nodemcu-build/) a try.
 
 ## Linux Build Environment
-NodeMCU firmware developers commit or contribute to the project on GitHub and might want to build their own full fledged build environment with the complete tool chain. There is a [post in the esp8266.com Wiki](http://www.esp8266.com/wiki/doku.php?id=toolchain#how_to_setup_a_vm_to_host_your_toolchain) that describes this.
\ No newline at end of file
+NodeMCU firmware developers commit or contribute to the project on GitHub and might want to build their own full fledged build environment with the complete tool chain. There is a [post in the esp8266.com Wiki](http://www.esp8266.com/wiki/doku.php?id=toolchain#how_to_setup_a_vm_to_host_your_toolchain) that describes this.
+
+## Customizing the build
+
+There is an optional file `user_config.h` that can be placed at the root of the build tree. If this file is present then it
```

Comment:
This file doesn't actually replace app/include/user_config.h -- it is used in addition to it. Now that I think about it, it is not a good name as it will cause that confusion. You can add modules here as well. I added a more complete example.

Maybe I should change the name to user_override.h -- what do you think?